pkgs/applications/networking/nordvpn/default.nix

```diff
@@ -0,0 +1,82 @@
+{ autoPatchelfHook, buildFHSEnv, dpkg, fetchurl, lib, stdenv, sysctl
+, iptables, iproute2, procps, cacert, libxml2, libidn2, zlib, wireguard-tools }:
+
+let
+  pname = "nordvpn";
+  version = "3.15.5";
+
+  nordVPNBase = stdenv.mkDerivation {
+    inherit pname version;
+
+    src = fetchurl {
+      url =
+        "https://repo.nordvpn.com/deb/nordvpn/debian/pool/main/nordvpn_${version}_amd64.deb";
+      hash = "sha256-7IvtpWiwVFaOc9/2NzG9mAGaWAeBZXNPdqCumdslawI=";
+    };
+
+    buildInputs = [ libxml2 libidn2 ];
+    nativeBuildInputs = [ dpkg autoPatchelfHook stdenv.cc.cc.lib ];
+
+    dontConfigure = true;
+    dontBuild = true;
+
+    unpackPhase = ''
+      runHook preUnpack
+      dpkg --fsys-tarfile $src | tar --extract
+      runHook postUnpack
+    '';
+
+    installPhase = ''
+      runHook preInstall
+      mv usr/ $out/
+      mv var/ $out/
+      mv etc/ $out/
+      runHook postInstall
+    '';
+  };
+
+  nordVPNfhs = buildFHSEnv {
+    name = "nordvpnd";
+    runScript = "nordvpnd";
+
+    # hardcoded path to /sbin/ip
+    targetPkgs = pkgs:
+      with pkgs; [
+        nordVPNBase
+        sysctl
+        iptables
+        iproute2
+        procps
+        cacert
+        libxml2
+        libidn2
+        zlib
+        wireguard-tools
+      ];
+  };
+
+in stdenv.mkDerivation {
```

Comment:
This derivation is based of [the ExpressVPN derivation](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/networking/expressvpn/default.nix), which uses the same pattern. How can they be merged together?

Reply:
Does merging the derivations together become easier if you build from source instead?

I tried my hand at modifying this PR to instead build from source. It still needs some work, but I'll post it here in case it can be of some help as a starting point

<details>
    <summary>nordvpn/default.nix</summary>

    ```nix
    { autoPatchelfHook, buildFHSEnvChroot, dpkg, fetchFromGitHub, lib, stdenv, sysctl
    , iptables, iproute2, procps, cacert, libxml2, libidn2, zlib, wireguard-tools, buildGoModule, pkg-config }:
    
    let
      pname = "nordvpn";
      version = "3.16.3";
    
      nordVPNCli = buildGoModule rec {
        inherit pname version;
    
        src = fetchFromGitHub {
          owner = "NordSecurity";
          repo = "nordvpn-linux";
          rev = "301b667871e0e6a29415d9a56bcf3c12b1efadc5";
          sha256 = "sha256-Kvoedrp1wpSGAZB0G6GZwp3lRYBNl6ZGuGBKkuz7nlE=";
        };
    
        vendorSha256 = "sha256-lLkx1tUSukOFFlFJh5qkfA0toYDnA3QdZLwhjjIaMPU=";
    
        CGO_CFLAGS = "-g -O2 -D_FORTIFY_SOURCE=2";
        CGO_LDFLAGS = "-Wl,-z,relro,-z,now";
        CGO_ENABLED = 1;
    
        LDFLAGS = ''
          -X main.Version= \
          -X main.Environment= \
          -X main.Hash= \
          -X main.Salt=
        '';
    
        buildInputs = [ libxml2 libidn2 ];
        nativeBuildInputs = [ dpkg autoPatchelfHook stdenv.cc.cc.lib ];
    
        buildPhase = ''
          go build \
            -buildmode=pie \
            -tags=drop,moose,telio \
            -ldflags "-s -w -linkmode=external ${LDFLAGS}" \
            -o "bin/nordvpn" "/build/source/cmd/cli"
        '';
      };
    
      nordVPNDaemon = buildGoModule rec {
        inherit pname version;
    
        src = fetchFromGitHub {
          owner = "NordSecurity";
          repo = "nordvpn-linux";
          rev = "301b667871e0e6a29415d9a56bcf3c12b1efadc5";
          sha256 = "sha256-Kvoedrp1wpSGAZB0G6GZwp3lRYBNl6ZGuGBKkuz7nlE=";
        };
    
        vendorSha256 = "sha256-lLkx1tUSukOFFlFJh5qkfA0toYDnA3QdZLwhjjIaMPU=";
    
        CGO_CFLAGS = "-g -O2 -D_FORTIFY_SOURCE=2";
        CGO_ENABLED = 1;
    
        LDFLAGS = ''
          -X main.Salt= \
          -X main.Version= \
          -X main.Environment= \
          -X main.Arch= \
          -X main.PackageType=
        '';
    
        buildInputs = [ libxml2 libidn2 ];
        nativeBuildInputs = [ dpkg autoPatchelfHook stdenv.cc.cc.lib pkg-config ];
    
        buildPhase = ''
          go build \
            -buildmode=pie \
            -tags=drop \
            -ldflags "-s -w -linkmode=external ${LDFLAGS}" \
            -o "bin/nordvpnd" "/build/source/cmd/daemon"
        '';
      };
    
      nordVPNfhs = buildFHSEnvChroot {
        name = "nordvpnd";
        runScript = "nordvpnd";
    
        # hardcoded path to /sbin/ip
        targetPkgs = pkgs:
          with pkgs; [
            nordVPNCli
            nordVPNDaemon
            sysctl
            iptables
            iproute2
            procps
            cacert
            libxml2
            libidn2
            zlib
            wireguard-tools
          ];
      };
    
    in stdenv.mkDerivation {
      inherit pname version;
    
      dontUnpack = true;
      dontConfigure = true;
      dontBuild = true;
    
      # DISCUSS: did we break the man pages?
      # DISCUSS: what comes from /var ?
      installPhase = ''
        runHook preInstall
        mkdir -p $out/bin $out/share
        ln -s ${nordVPNCli}/bin/nordvpn $out/bin
        ln -s ${nordVPNfhs}/bin/nordvpnd $out/bin
        ln -s ${nordVPNCli}/share/* $out/share/
        ln -s ${nordVPNCli}/var $out/
        runHook postInstall
      '';
    
      meta = with lib; {
        description = "CLI client for NordVPN";
        homepage = "https://www.nordvpn.com";
        license = licenses.gpl3Only;
        maintainers = with maintainers; [ LuisChDev ];
        platforms = [ "aarch64-linux" "x86_64-linux" ];
      };
    }    
    ```

</details>